cloud/pkg/admissioncontroller/controller/admission.go

```diff
@@ -0,0 +1,255 @@
+package controller
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/kubeedge/beehive/pkg/common/log"
+	"github.com/kubeedge/kubeedge/cloud/pkg/admissioncontroller/constants"
+	"github.com/kubeedge/kubeedge/cloud/pkg/admissioncontroller/utils"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+)
+
+var scheme = runtime.NewScheme()
+var codecs = serializer.NewCodecFactory(scheme)
+
+func init() {
+	addToScheme(scheme)
+}
+
+func addToScheme(scheme *runtime.Scheme) {
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(admissionv1beta1.AddToScheme(scheme))
+	utilruntime.Must(admissionregistrationv1beta1.AddToScheme(scheme))
+}
+
+// AdmissionController implements the admission webhook for validation of configuration.
+type AdmissionController struct {
+	Client *kubernetes.Clientset
+}
+
+func strPtr(s string) *string { return &s }
+
+// Register registers the admission webhook.
+// fixme: plugable?
+func (ac *AdmissionController) register(WebhookName string, context *utils.CertContext) error {
+	log.LOGGER.Warn("Get into register")
```

Comment:
Is this log required ?

Reply:
doesn't really, it only for debug purpose, I have removed those logs.